### PR TITLE
Bug fixed: File > Close Data Book

### LINKED ISSUE #7475: Minor Bug with File > Close Data Book
During the KMD workshop, we noticed that File > Close Data Book seems to not work in the following cases.

1. When the Filter is active and try to `Close Data Book` and it seems to not work
2. After saving a dataset with `File > Save As > Save Data As` then try to `Close Data Book` same as 1), it doesn't close the databook
image

### PR DESCRIPTION
**Pull request by N-thony**
Fixes #7475
@lloyddewit as discussed this bug was due to the fact the check when closing the databook was not done correctly, please take a look.
@africanmathsinitiative/developers this is ready for review.

### CODE REVIEW

File: instat/frmMain.vb [hunk 1]
```diff
@@ -71,7 +71,7 @@ Public Class frmMain
     ''' it's set to false by ucrDataView control when state of data has been changed
     ''' it's set to true by dlgSaveAs dialog and save menu when data has been successfully saved 
     ''' </summary>
-    Public Property bDataSaved As Boolean = False
+    Public Property bDataSaved As Boolean
```
Comment (lloyddewit) [inline instat/frmMain.vb:74]: In VB dotnet, the default value for a boolean is False. 
Removing the assignment does not change the behaviour but it makes the code less readable.
Undo this change?

File: instat/frmMain.vb [hunk 2]
```diff
@@ -1566,13 +1566,13 @@ Public Class frmMain
...
-                strSaveFilePath = ""
-            End If
+        If (Not bDataSaved AndAlso ucrDataViewer.GetWorkSheetCount() > 0 _
+                            AndAlso DialogResult.Yes = MsgBox("Are you sure you want to close your data?" &
+                                    Environment.NewLine & "Any unsaved changes will be lost.",
+                                    MessageBoxButtons.YesNo, "Close Data")) _
+                            OrElse bDataSaved Then
+            clsRLink.CloseData()
+            strSaveFilePath = ""
         End If
```
Comment (lloyddewit) [inline instat/frmMain.vb:1576]: Is something like this more readable?

```suggestion
        If ucrDataViewer.GetWorkSheetCount() <= 0 Then
            Exit Sub
        End If

        If Not bDataSaved _
                AndAlso DialogResult.No = MsgBox("Are you sure you want to close your data?" &
                                    Environment.NewLine & "Any unsaved changes will be lost.",
                                    MessageBoxButtons.YesNo, "Close Data") Then
            Exit Sub
        End If

        clsRLink.CloseData()
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Comment by N-thony** [comment]
Thanks @lloyddewit, I resolved your comment.

**Comment by lloyddewit** [comment]
@rdstern if you can test/approve, then we can merge, thanks

**Review by rdstern — CHANGES_REQUESTED** [review]
@N-thony it now works - but perhaps too well!
The previous Close - when it worked - also gave the message:
image

Now it seems to close fine, but without the message.  I rather liked the warning we had before.

**Comment by N-thony** [comment]
> @N-thony it now works - but perhaps too well! The previous Close - when it worked - also gave the message: image
> 
> Now it seems to close fine, but without the message. I rather liked the warning we had before.

@rdstern the message is still there but only when the data are not saved, otherwise it will close without a message.
Do we still need to give a message even if the data are already saved and there is no changes?

**Review by rdstern — APPROVED** [review]
Good response @N-thony and you are correct!  Well done.  That's better!

**Pull request merged**